### PR TITLE
[13.0][IMP] sale_order_product_assortment: Use product_assortment domain and only one field to get allowed products result from whitelist and blacklist

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT30584
ping @carlosdauden @joao-p-marques 

This PR depends on:
* https://github.com/OCA/product-attribute/pull/878

### CODE REVIEW

File: sale_order_product_assortment/models/sale_order.py [hunk 1]
```diff
@@ -6,52 +6,31 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    whitelist_product_ids = fields.Many2many(
+    allowed_product_ids = fields.Many2many(
```
Comment (joao-p-marques) [inline sale_order_product_assortment/models/sale_order.py:9]: Would a migration script be needed to rename this table and keep the data?

**Review by joao-p-marques — COMMENTED** [review]
Code review

File: sale_order_product_assortment/models/sale_order.py (hunk 1 above)
Comment (sergio-teruel) [reply sale_order_product_assortment/models/sale_order.py:9]: No because this field is computed no store.

Comment (joao-p-marques) [reply sale_order_product_assortment/models/sale_order.py:9]: Oh right, sorry :man_facepalming:

**Review by joao-p-marques — APPROVED** [review]
Code review :+1:

**Comment by pedrobaeza** [comment]
Please check tests

**Comment by sergio-teruel** [comment]
This PR has dependencies

File: sale_order_product_assortment/models/sale_order.py [hunk 2]
```diff
@@ -6,52 +6,34 @@
...
+        if self[partner_field]:
+            filters = IrFilters.browse()
+            filters_partner_domain = self.env["ir.filters"].search(
+                [("is_assortment", "=", True)]
             )
+            for ir_filter in filters_partner_domain:
+                if self[partner_field] & ir_filter.all_partner_ids:
+                    filters |= ir_filter
+                    self.allowed_product_ids += self.env["product.product"].search(
+                        ir_filter._get_eval_domain()
```
Comment (carlosdauden) [inline sale_order_product_assortment/models/sale_order.py:36]: We can add sale_ok=True to keep odoo standar behaviour

Comment (pedrobaeza) [reply sale_order_product_assortment/models/sale_order.py:36]: I have thought about this, but if added, you will see different results when previewing the filter results, and when searching for it in the sales order, so for sure that will confuse user. Do you think that's correct?

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@sbidoul working!

**Comment by sbidoul** [comment]
Cool. I gave it a new token.

**Pull request merged**